proj/src/main/webapp/db.js

```diff
@@ -0,0 +1,302 @@
+        if (event.keyCode === 13) { // 13 is the keycode for the enter key
+            pushChatMessage();
+        }
+    });
+}
+
+function createMessageWithTemplate(key, messageObj) {
+    const messageTemplate = document.getElementById('message-temp');
+    const docFrag = messageTemplate.content.cloneNode(true);
+    const message = docFrag.querySelector('.message')
+            
+    const msgBody = message.querySelector('.message-body');
+    msgBody.innerText = messageObj.content;
+
+    const timestamp = message.querySelector('#timestamp');
+    timestamp.innerText = new Date(messageObj.timestamp).toLocaleString();
+
+    message.id = key;
+    addUsernameToMessage(messageObj.senderUID, message)
+    return message;
+}
+
+
+
+function getDbRef(tag, chatId) {
+    const path = "/chat/"+tag+"/"+chatId+"/messages";
+    const dbRefObj = firebase.database().ref(path);
+    return dbRefObj;
+}
+
+async function makePreviewWithLastMessage(tag, chatId) {
+    const tagRefObj = firebase.database().ref('/chat/'+tag+'/'+chatId+'/chatInfo')
+    await tagRefObj.on('value', async function (snap) {
+
+        const preview = await makeChatPreview(snap.val(), tag, chatId);
+
+        const sidebar = document.getElementById('chats-submenu');
+        sidebar.prepend(preview);
+    });
+}
+
+async function makeChatPreview(chatInfoObj, tag, chatId) {
+    if (document.getElementById(chatId)) {
+        oldPreview = document.getElementById(chatId);
+        oldPreview.remove();
+    }
+
+    const previewTemplate = document.getElementById('chat-preview-temp');
+    const docFrag = previewTemplate.content.cloneNode(true);
+    const preview = docFrag.querySelector(".chat-preview")
+
+    const chatName = preview.querySelector('#chat-name');
+    chatName.innerText = chatInfoObj.name;
+    const msgBody = preview.querySelector('.message-body');
+    msgBody.innerText = chatInfoObj.lastMessage;
+
+    const timestamp = preview.querySelector('#timestamp');
+    timestamp.innerText = new Date(chatInfoObj.timestamp).toLocaleString();
+    
+    preview.setAttribute("id", chatId)
+    changeChatOnClick(preview, tag, chatId);
+    await addUsernameToMessage(chatInfoObj.lastAuthor, preview);
+
+    return preview;
+}
+
+// i can't access two paths at the same time so i need two separate functions :/
+async function addUsernameToMessage(uid, preview) {
+    const userRef = firebase.database().ref('/users/'+uid);
+    await userRef.once('value', snap => {
+        preview.querySelector('#username').innerText = snap.val().firstName + ' ' + snap.val().lastName;
+    })
+}
+
+function initBio() {
+    const bioBox = document.getElementById('user-bio');
+    const editInputBox = document.getElementById('bio-edit');
+
+    bioBox.addEventListener('dblclick', function() {
+        this.hidden = true;
+
+        editInputBox.hidden = false;
+        editInputBox.value = this.innerText
+
+        editInputBox.focus();
```

Comment:
i am not sure if this is an updated function but in the current version, when you double click the bio, the bio content is hidden

Reply:
nvm i just saw you commented on the issue